Docstring updates: load_tables, simplify_tables, EdgeTable

ashander opened the pull request:
contributing to #323

ashander commented:
looks like I'll need to add some tests to load_tables.

jeromekelleher commented on msprime/tables.py:
```diff
@@ -603,15 +600,14 @@ def sort_tables(*args, **kwargs):
-    :param SiteTable sites:
-    :param MutationTable mutations:
-    :param int edge_start: The index in the edge table where sorting starts.
+    :param NodeTable nodes (required).
```
I don't this this is valid sphinx markup. It should be something like
```
:param NodeTable nodes: The tree sequence nodes (required).
```
See http://www.sphinx-doc.org/en/stable/domains.html for examples of the required markup formats.

jeromekelleher commented on msprime/tables.py:
```diff
@@ -649,6 +646,7 @@ def simplify_tables(*args, **kwargs):
     :param MutationTable mutations: The MutationTable to be simplified.
     :param bool filter_invariant_sites: Whether to remove sites that have no
         mutations from the output (default: True).
+    :param sequence_length: The length of the sequence.
```
Need a type here (float I guess?)

jeromekelleher commented on msprime/trees.py:
```diff
@@ -818,8 +818,41 @@ def load(path):
+
+    :param NodeTable nodes: The NodeTable.
+    :param EdgeTable edges: The EdgeTable.
+    :param sequence_length: The length of the sequence.
```
Type missing.

jeromekelleher commented on msprime/trees.py:
```diff
@@ -818,8 +818,41 @@ def load(path):
+    :return: A :class:`msprime.TreeSequence` consistent with the tables.
+    :rtype: TreeSequence
+    """
+    if nodes is None:
```
I don't think there's much point in doing these checks here, as it's all done already by the low-level module. I think it's better to just stick with the existing way of packing the (*args, **kwargs) --- otherwise, we're doing the same thing twice in two places.

jeromekelleher commented:
Actually, I just spotted the code changes to sort_tables, and I think we should revert these before merging. I realise the *args, **kwargs approach isn't ideal for Sphinx documentation, but we can figure out some workaround for this.

ashander commented:
Made these changes and built docs locally without issues. Thanks!

ashander commented:
Oops, I squashed but then just realized we have an existing solution in sphinx to list the arguments for `*args, **kwargs` functions. I updated that just now for simplify_tables

jeromekelleher commented:
Merging. Thanks @ashander!
